### PR TITLE
chore: Oauth temporary ux improvements

### PR DESCRIPTION
**Pull request by amk-dev**
Fixes HFE-382

**Changes**

1. Disable the OpenID Connect Discovery URL when Authorization/Access Token URLs are present.
2. Disable Authorization/Access Token URLs when the OpenID Connect Discovery URL is present.
3. Implement OAuth fetch calls through interceptors.
4. Fix an if statement that made OpenID Connect discovery URLs mandatory

**Note**
We will be removing OpenID Connect Discovery URL very soon. this is temporary solution till then.

### CODE REVIEW

**Comment by anwarulislam** [comment]
@amk-dev there is an issue you can see in the video. I think we need to handle empty value here.


https://github.com/hoppscotch/hoppscotch/assets/26295990/5ab58c80-7ad5-4cac-b0fc-8f83e1950d79

**Review by anwarulislam — CHANGES_REQUESTED** [review]
@amk-dev there is an issue you can see in the video. I think we need to handle empty value here.


https://github.com/hoppscotch/hoppscotch/assets/26295990/5ab58c80-7ad5-4cac-b0fc-8f83e1950d79

File: packages/hoppscotch-common/src/helpers/oauth.ts [hunk 1]
```diff
@@ -309,4 +296,32 @@ const clearPKCEState = () => {
...
+async function runRequestThroughInterceptor(config: AxiosRequestConfig) {
+  if (interceptorService.currentInterceptorID.value === "extension") {
+    // wait till the extension resolves
+    await until(extensionService.extensionStatus).toMatch(
+      (status) => status !== "waiting",
+      {
+        timeout: 1000,
+      }
+    )
+  }
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/helpers/oauth.ts:308]: I would recommend to keep specific Interceptor logic out of the general app code.

What if we do this wait within `ExtensionInterceptorService`'s `runRequest` call ?

**Review by AndrewBastin — CHANGES_REQUESTED** [review]

File: packages/hoppscotch-common/src/helpers/oauth.ts (hunk 1 above)
Comment (amk-dev) [reply packages/hoppscotch-common/src/helpers/oauth.ts]: yeah. that would work.

**Review by anwarulislam — APPROVED** [review]
LGTM 🚀

**Review by AndrewBastin — APPROVED** [review]
lgtm

**Pull request merged**